### PR TITLE
[LEVWEB-1376] Add timezone to search

### PR DESCRIPTION
**Pull request by aarongill1**

### CODE REVIEW

File: src/lib/db/query.js [hunk 1]
```diff
@@ -10,9 +11,9 @@ const groupByType = ' GROUP BY dataset';
 const groupByDateTypeUser = ' GROUP BY date_time::date, dataset, username ORDER BY date_time::date';
 const groupByTypeGroup = 'GROUP BY name, dataset';
 const totalCount = 'SELECT count(*)::INTEGER FROM lev_audit';
-const forToday = ' WHERE date_time::DATE = current_date';
-const fromDate = 'date_time::DATE >= $(from)';
-const toDate = 'date_time::DATE < $(to)';
+const forToday = ` WHERE (CURRENT_TIMESTAMP ${timeZone})::DATE = (date_time ${timeZone})::DATE;`;
```
Comment (RobinKnipe) [inline src/lib/db/query.js:14]: It looks like we have a lot of `::DATE` truncation of `date_time`, i.e. everywhere the column is used, so the following might be useful:
```js
const DTZ = `(date_time ${timeZone})::DATE`;
```

Comment (aarongill1) [reply src/lib/db/query.js]: Refactored

File: src/lib/db/query.js [hunk 2]
```diff
@@ -3,30 +3,33 @@
...
 
 const timeZone = 'AT TIME ZONE \'europe/london\'';
-const countsByType = 'SELECT dataset, count(*)::INTEGER FROM lev_audit WHERE date_time > $(from)';
+const DTZ = `(date_time ${timeZone})::DATE`;
+const countsByType = `SELECT dataset, count(*)::INTEGER FROM lev_audit WHERE ${DTZ} > $(from)`;
 const countsByUser =
-  'SELECT date_time::DATE AS date, dataset, username, count(*)::INTEGER FROM lev_audit WHERE date_time > $(from)';
-const until = 'AND date_time < $(to)';
+  `SELECT ${DTZ} AS date, dataset, username, count(*)::INTEGER 
+  FROM lev_audit WHERE ${DTZ} > $(from)`;
```
Comment (RobinKnipe) [inline src/lib/db/query.js:10]: Any reason this query is split over 2 lines?

File: src/lib/db/query.js [hunk 3]
```diff
@@ -3,30 +3,33 @@
...
+const until = `AND ${DTZ} < $(to)`;
 const groupByType = ' GROUP BY dataset';
-const groupByDateTypeUser = ' GROUP BY date_time::date, dataset, username ORDER BY date_time::date';
+const groupByDateTypeUser =
+  ` GROUP BY ${DTZ}, dataset, username ORDER BY ${DTZ}`;
```
Comment (RobinKnipe) [inline src/lib/db/query.js:14]: Any reason this statement is split over 2 lines?

File: src/lib/db/query.js [hunk 4]
```diff
@@ -44,10 +47,10 @@ const sqlBuilder = (obj) => {
 module.exports = {
   usageByDateType: (from, to, group) => db.manyOrNone(
     sqlBuilder({
-      'SELECT': 'date_time::DATE AS date, dataset, count(*)::INTEGER',
+      'SELECT': `${DTZ} AS date, dataset, count(*)::INTEGER`,
       'FROM': 'lev_audit',
       'WHERE': [from && fromDate, to && toDate, group && searchGroup],
-      'GROUP BY': 'date_time::date, dataset ORDER BY date_time::date'
+      'GROUP BY': `${DTZ}, dataset ORDER BY ${DTZ}`
```
Comment (RobinKnipe) [inline src/lib/db/query.js:53]: You don't need to use the `DTZ` here, and in fact it's probably not good practise... instead, you should be able to do the following:
```sql
GROUP BY 1, 2 ORDER BY 1
```

File: src/lib/db/query.js (hunk 2 above)
Comment (aarongill1) [reply src/lib/db/query.js:10]: Was because of linting error but after refactor, no. Have changed.

File: src/lib/db/query.js (hunk 4 above)
Comment (aarongill1) [reply src/lib/db/query.js:53]: OK done

**Pull request merged**